pkg/pubsub/pubsub_init.go

```diff
@@ -0,0 +1,95 @@
+package pubsub
+
+import (
+	"context"
+	"fmt"
+
+	psClient "github.com/open-policy-agent/gatekeeper/pkg/pubsub/client"
+	"github.com/open-policy-agent/gatekeeper/pkg/util"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeClient "sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var log = logf.Log.WithName("pubsub")
+
+var _ manager.Runnable = &runner{}
+
+type runner struct {
+	mgr manager.Manager
+}
+
+func AddToManager(m manager.Manager) error {
+	mr := new(m)
+	return m.Add(mr)
+}
+
+func new(mgr manager.Manager) *runner {
+	mr := &runner{
+		mgr: mgr,
+	}
+	return mr
+}
+
+var initiatedTools []psClient.PubSub
+
+// Start implements the Runnable interface.
+func (r *runner) Start(ctx context.Context) error {
```

Comment:
Per elsewhere, we should not rely on manager here, or pubsub may not work in non-K8s environments, like G8r test

Reply:
Right now client initialization is within the reconciler if we want to integrate into g8t test, additional exclusive function for g8r test makes sense to me. wdyt?

Reply:
It does not make sense:

* right now there is no reconcile loop. A reconcile loop is inside of a controller (which you do not have). I think a controller is a good idea. It should watch whatever config objects we use (config maps, CRDs, whatever) and configure the pubsub system accordingly. In that case, the pubsub system would be injected into the reconciler.

* "an additional exclusive function" sounds to me like the code is not shareable across systems, which means we'd be copy/pasting functionality. The code should be shareable. Mutation, expansion, and the constraint frameworks are all models that do this (work both as controllers and with `gator test`), so there are models you can use to get this to work.

Reply:
I looked into mutation and I think I get what you mean with this and [other](https://github.com/open-policy-agent/gatekeeper/pull/2538#discussion_r1109181444) comment. I implemented a similar approach for pub-sub to make it generic and work with controller and `gator` as well (i.e without depending on kube client to initiate pub sub system).